### DISCUSSION TITLE
ChatComposite with react-native

### DISCUSSION THREAD

**Discussion by tariromukute** [Azure/communication-ui-library#3803] [Q&A]
I am trying to use react-native with ACS. There is an example for ACS calling feature with react-native [here](https://github.com/Azure-Samples/communication-services-ui-library-react-native) where `RNAzureCommunicationUICalling` module from your React Native Javascript code is used to implement ACS with react-native.

I want to implement the ChatComposite. I can't see any reference for the ChatComposite. Is ChatComposite support for react-native? If so, where can I find resources or documents to help me get started?

**Comment by jpeng-ms** [comment]
Hi @tariromukute , thanks for you question, this is John and I will be helping you this week. Please allow me to check with our team and I will get you back shortly.

**Comment by jpeng-ms** [comment]
Hi @tariromukute , thanks for your patience. I have checked with our team and you are correct that we currently only have a sample for `CallComposite` in React Native. That being said, we welcome you to raise a feature request in this forum: https://feedback.azure.com/d365community/forum/81ff6d2b-0c25-ec11-b6e6-000d3a4f0858.

Alternatively, our `ChatComposite` is available in public preview for native platforms such as [iOS](https://github.com/Azure/communication-ui-library-ios/tree/main/AzureCommunicationUI/sdk/AzureCommunicationUIChat) and [Android](https://github.com/Azure/communication-ui-library-android/tree/main/azure-communication-ui/chat). If you would like, you can explore creating the bindings by following the documentations from React Native. 

Hope this helps. Please let me know if you have any more questions!

**Comment by tariromukute** [comment]
Hi @jpeng-ms, thanks for checking in with the team, and for the suggestion of creating bindings. I am new to React Native and I wanted a quick POC for a chat service. I followed your suggestion of bindings and used the tutorials [Swift in React Native - The Ultimate Guide Part 2: UI Components](https://teabreak.e-spres-oh.com/swift-in-react-native-the-ultimate-guide-part-2-ui-components-907767123d9e) and [React-Native UI Native components in Swift and Java](https://www.admios.com/blog/react-native-ui-native-components-in-swift-and-java) to port the ChatComposite, at the moment only managed to do so for iOS.

I saw that react native has something called _Fabric Native Components_ in their new architecture. Since this is still experimental I was hesitate on going that route as a first timer, however it seems like it has the advantage that you can export the components to be reused by others. If you don't have any work on this maybe I can raise a feature request for that instead.

**Reply by jpeng-ms** [reply]
Hi @tariromukute, we currently don't have any plans on "Fabric Native Components", feel free to raise a feature request for us.

Thanks again!
